Assets/Scripts/EnemyController.cs

```diff
@@ -65,24 +71,56 @@ void Update()
         {
             if (currentHealth < 0)
             {
-                animator.SetBool("IsDead", true);
-                PhotonNetwork.Destroy(gameObject);  //TODO: Replace with running away logic. Only destroy when the exit point(fountain of color) is reached.
+                photonView.RPC("Blobify", RpcTarget.All);
+                //  //TODO: Replace with running away logic. Only destroy when the exit point(fountain of color) is reached.
+            }
+            if(isBlobified && distanceToKeyLocationToDespawn > Vector3.Distance(GetNearestKeyLocation().position, transform.position))
+            {
+                PhotonNetwork.Destroy(gameObject);
             }
         }
-        FindNavTarget();
-        distanceToPlayer = Vector3.Distance(player.position, transform.position);
-        if (distanceToPlayer <= minDistForMovement)
+        if (!isBlobified)
         {
-            SetSpeed(speed);
-            if (isAttackReady)
+            FindNavTarget();
+            distanceToPlayer = Vector3.Distance(player.position, transform.position);
+            if (distanceToPlayer <= minDistForMovement)
             {
-                StartCoroutine(AttackPlayer());
+                SetSpeed(speed);
+                if (isAttackReady)
+                {
+                    StartCoroutine(AttackPlayer());
+                }
+            }
+            else
+            {
+                SetSpeed(0);
             }
         }
-        else
+    }
+
+    [PunRPC]
+    void Blobify()
+    {
+        animator.SetBool("IsDead", true);
+        isBlobified = true;
+        agent.stoppingDistance = 0;
+        agent.destination = GetNearestKeyLocation().position;
+        SetSpeed(speed);
+        //TODO?: set color to nice pink
+    }
+
+    Transform GetNearestKeyLocation()
+    {
+        GameObject[] keyLocations = GameObject.FindGameObjectsWithTag("KeyLocation");
```

Comment:
you can get this at the start, so you don't have to constantly find the same objects every time

Reply:
What if the enemy follows player to a different key location? It should always find the nearest one

Reply:
I meant the object array you can get at the start